addons/payment_razorpay/tests/test_payment_provider.py

```diff
@@ -24,3 +25,27 @@ def test_signature_calculation_for_redirect_data(self):
         self.assertEqual(
             calculated_signature, '437b72e4e87362a39951b44487cf698410b074afdbed19ec44fffd32d2f863f3'
         )
+
+    def test_credentials_constraints(self):
+        """ Test that enabling a Razorpay provider without credentials raises a ValidationError,
+        and with credentials it succeeds.
+        """
+        provider_no_credentials = self.env['payment.provider'].create({
+            'name': 'Razorpay Missing Credentials',
+            'code': 'razorpay',
+            'state': 'test',
+        })
+
+        with self.assertRaises(ValidationError), self.cr.savepoint():
+            provider_no_credentials.write({'state': 'enabled'})
+
+        provider_valid_creds = self.env['payment.provider'].create({
+            'name': 'Razorpay Valid Creds',
+            'code': 'razorpay',
+            'razorpay_key_id': 'test_key',
+            'razorpay_key_secret': 'test_secret',
+            'state': 'test',
+        })
+
+        provider_valid_creds.write({'state': 'enabled'})
+        self.assertEqual(provider_valid_creds.state, 'enabled')
```

Comment:
To test as close to the covered code as possible, use `_assert_does_not_raise()`, similar to how you used `assertRaises`.